Sources/WhisperKit/Core/TranscribeTask.swift

```diff
@@ -0,0 +1,349 @@
+//  For licensing see accompanying LICENSE.md file.
+//  Copyright © 2024 Argmax, Inc. All rights reserved.
+
+import Foundation
+import OSLog
+import CoreML
+
+private let logger = Logger(
+    subsystem: Constants.Logging.subsystem,
+    category: "TranscribeTask"
+)
+private let signposter = OSSignposter(logger: logger)
+
+/// Responsible for transcribing audio chunk to text using the provided models and configurations.
+@available(macOS 13, iOS 16, watchOS 10, visionOS 1, *)
+final class TranscribeTask {
+    private let maxTokenContext: Int
+    private var timings: TranscriptionTimings
+    private let progress: Progress
+    private let audioEncoder: any AudioEncoding
+    private let featureExtractor: any FeatureExtracting
+    private let segmentSeeker: any SegmentSeeking
+    private let textDecoder: any TextDecoding
+    private let tokenizer: any WhisperTokenizer
+
+    init(
+        maxTokenContext: Int,
+        currentTimings: TranscriptionTimings,
+        progress: Progress?,
```

Comment:
`Progress` [should be thread safe](https://oleb.net/blog/2014/03/nsprogress/) so I think we can pass it from the upstream